public_html/wp-content/mu-plugins/blocks/assets/src/shared/avatar/index.js

```diff
@@ -14,6 +14,13 @@ function AvatarImage( {
 	name,
 	size,
 	url,
+	// this feels like a weird way to declare default arguments, mostly from a readability standpoint. i wonder if there's a better way, but i'm not really sure.
+	// i don't think i've seen it done this way in gutenberg, but i could be wrong
+	// maybe this should extend Component so it could simply accept a `props` object and then have a `defaultProps` property?
+		// or is this some es6 way of picking specific things out of the props that are already being passed in?
+	// or maybe have a statement where it merges the passed props with the defined defaults or something?
+	// or maybe just put it all on 1 line?
+	// if we change it here, we should change all the other instances as well
```

Comment:
I got this pattern from some other component in Gutenberg, but I'm not sure which one. Unfortunately, as I discovered over the course of building this, Gutenberg does not have a consistent way of defining components or blocks, probably due to both varying contributor styles and the fact that the codebase was developed over more than a year, during which a lot of best practices in React and es6 probably changed.

I agree that we should standardize, though, and I think extending `Component` is a good way to do that.

Reply:
@iandunn I dug into this a bit more, and I think I have a better explanation now. I guess I'm not entirely sure which parts of this feel "weird" to you, but here's what I've got:

* This is actually a fairly common way to "destructure" function parameters in es6. They aren't really defaults in this case, because we're not assigning default values to them, just declaring them. [Here's one post that discusses this approach.](https://simonsmith.io/destructuring-objects-as-function-parameters-in-es6/)
* Declaring the component as a "pure function" instead of a class extending `Component` is considered by some to be a best practice in the case that the component doesn't have any modifiable state. If the component did have state and/or need to use lifecycle methods, then it would make sense to extend the `Component` class. Some discussion [here](https://stackoverflow.com/questions/36097965/react-when-to-use-es6-class-based-components-vs-functional-es6-components) and [here](https://code.tutsplus.com/tutorials/stateful-vs-stateless-functional-components-in-react--cms-29541).

Given these points, most of our components could probably actually be functions instead of classes...

Reply:
There might be some minor performance efficiencies when using stateless function components, but I agree, it's probably not worth refactoring existing stuff. Gutenberg's coding guidelines actually mention transitioning function components to classes when state needs to be introduced, so maybe we could take a similar approach:

https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#class-definition

Reply:
Ah, thanks for those links. I remove my objections :)

For cases like this where the list of properties is short, and we're not assigning values, I think it still might be nice to put it all on one line, though. If either of those is false, though, then I agree it's more readable to use separate lines. I'm also fine using separate lines if you think the consistency is better. Not a big deal either way.

I also like the idea of using pure functions in cases where we don't need state/lifecycle.